bark/world/prediction/prediction_settings.cpp

```diff
@@ -7,30 +7,70 @@
 // For a copy, see <https://opensource.org/licenses/MIT>.
 
 #include "bark/world/prediction/prediction_settings.hpp"
+#include "bark/models/behavior/not_started/behavior_not_started.hpp"
 #include "bark/world/observed_world.hpp"
+// #include "bark/models/behavior/constant_velocity/constant_velocity.hpp"
 
 namespace bark {
 namespace world {
 namespace prediction {
 
 using bark::models::behavior::BehaviorModelPtr;
+using bark::models::behavior::BehaviorNotStarted;
+using models::behavior::BehaviorStatus;
 
 PredictionSettings::PredictionSettings(
-    const BehaviorModelPtr& ego_prediction,
-    const BehaviorModelPtr& others_prediction)
-    : ego_prediction_model_(ego_prediction),
-      others_prediction_model_(others_prediction) {}
+    const BehaviorModelPtr& ego_prediction_model,
+    const BehaviorModelPtr& default_prediction_model,
+    const BehaviorModelPtr& specific_prediction_model,
+    const std::vector<AgentId>& specific_prediction_agents)
+    : ego_prediction_model_(ego_prediction_model),
+      specific_prediction_model_(specific_prediction_model),
+      default_prediction_model_(default_prediction_model),
+      specific_prediction_agents_(specific_prediction_agents.begin(),
+                                  specific_prediction_agents.end()) {}
 
 void PredictionSettings::ApplySettings(
     bark::world::ObservedWorld& observed_world) const {
-  observed_world.SetEgoBehaviorModel(
-      BehaviorModelPtr(ego_prediction_model_->Clone()));
-  for (const auto& agent : observed_world.GetOtherAgents()) {
-    observed_world.SetBehaviorModel(
-        agent.first, BehaviorModelPtr(others_prediction_model_->Clone()));
+  // TODO: Clean up HACK to keep Behavior Status
```

Comment:
now we need to specify several agent models?
what is specific_prediction_model_?
Could you clarify this?

Reply:
Prediction settings now allow to set behavior models for the ego vehicle, a default prediction model for the other vehicle, and a specific prediction for some agent ids. The feature about the specific prediction model was added. It can be used for a more detailed model for nearby agents. Or to let the multi-agent mcts only plan for nearby agents, and predict the others with some default model

Reply:
okay, but should we not have a constructor for the case we do not need a specific prediction  model?
E.g. overload the constructor

Reply:
I implemented it